pkg/api/types.go

```diff
@@ -403,6 +404,13 @@ const (
 	PodUnknown PodPhase = "Unknown"
 )
 
+type PodCondition string
+
+// These are the valid service statuses of pods.
+const (
+	PodInService = "InService"
```

Comment:
And:
```
type PodConditionStatus string
const (
	PodConditionFull    PodConditionStatus = "Full"
	PodConditionNone    PodConditionStatus = "None"
	PodConditionUnknown PodConditionStatus = "Unknown"
)
```
With pods degraded/partial states make a little less sense than for nodes, so I might be willing to use Good/Bad/Unknown or something similar. However, an example of a Degraded/Partial condition might be some containers in the pod reporting ready and some unresponsive, and/or some of the last N probes failing but not yet at the threshold to declare it not ready (if/when we add thresholds).